tesseract/tesseract_planning/tesseract_motion_planners/include/tesseract_motion_planners/trajopt/trajopt_motion_planner.h

```diff
@@ -54,7 +54,7 @@ class TrajOptMotionPlanner : public MotionPlanner
    * @param config The planners configuration
    * @return True if successful otherwise false
    */
-  bool setConfiguration(const TrajOptPlannerConfig::Ptr& config);
+  bool setConfiguration(const TrajOptPlannerConfig::Ptr config);
```

Comment:
It may be good to remove the const so it can be moved into config_.

Reply:
Ok. All of the discussion about RVO before got me curious. I put this into godbolt as well. All 4 of these seem to do the same in GCC 9.2. Maybe I'm doing something wrong
```
    void setBarMember1(const std::shared_ptr<Bar> input_bar) {bar_member = input_bar;}
    void setBarMember2(const std::shared_ptr<Bar> input_bar) {bar_member = std::move(input_bar);}
    void setBarMember3(std::shared_ptr<Bar> input_bar) {bar_member = input_bar;}
    void setBarMember4(std::shared_ptr<Bar> input_bar) {bar_member = std::move(input_bar);}
```
[move_vs_assign_in_setMember.txt](https://github.com/ros-industrial-consortium/tesseract/files/3794158/move_vs_assign_in_setMember.txt)

Reply:
I think that is because in gcc 9.2 uses c++17 with guaranteed NRVO (named return value optimization) which is probably why you get the same results. If you add the compiler warnings below you most likely would get a warning that the std::move is redundant. 

-Wpessimizing-move -Wredundant-move

Reply:
Though prior to c++17 this was not the case.

Reply:
I added those options, but I don't see where compiler warnings are printed. In the compiler output it just says "compiler returned 0". I was under the impression that most major compilers already did NRVO but C++17 just added it to the spec. It looks like GCC 9 is available in Ubuntu 19, not 16, or 18. 

Regardless, I'll follow your recommendation. What would you like me to do?

Reply:
Actually I think what you have is good.